pytorch_lightning/utilities/reproducibility.py

```diff
@@ -0,0 +1,21 @@
+"""Helper functions to help with reproducibility of models """
+import os
+
+import numpy as np
+import random
+import torch
+
+
+def seed_everything(seed: int = None):
+    """Function that sets seed for pseudo-random number generators  """
+    try:
+        seed = int(seed)
+    except (TypeError, ValueError):
+        return
+    random.seed(seed)
+    os.environ["PYTHONHASHSEED"] = str(seed)
+    np.random.seed(seed)
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
+    torch.backends.cudnn.deterministic = True
```

Comment:
IMO this shouldn't be part of a seed function. And I think there is also a trainer arg handling this IIRC

Reply:
Probably you are saying about `benchmark` flag, but probably it is a good idea. I put it as a `Trainer` flag, and added few warnings if seed is set and deterministic is not